src/app/clusters/color-control-server/color-control-server.cpp

```diff
@@ -25,11 +25,303 @@
+        }
+
+        if (ColorControlServer::Instance().HasFeature(endpoint, ColorControlServer::Feature::kHueAndSaturation))
+        {
+            uint8_t saturationValue = 0x00;
+            Attributes::CurrentSaturation::Get(endpoint, &saturationValue);
+            AddAttributeValuePair(pairs, Attributes::CurrentSaturation::Id, saturationValue, attributeCount);
+        }
+
+        if (ColorControlServer::Instance().HasFeature(endpoint, ColorControlServer::Feature::kColorLoop))
+        {
+            uint8_t loopActiveValue = 0x00;
+            Attributes::ColorLoopActive::Get(endpoint, &loopActiveValue);
+            AddAttributeValuePair(pairs, Attributes::ColorLoopActive::Id, loopActiveValue, attributeCount);
+
+            uint8_t loopDirectionValue = 0x00;
+            Attributes::ColorLoopDirection::Get(endpoint, &loopDirectionValue);
+            AddAttributeValuePair(pairs, Attributes::ColorLoopDirection::Id, loopDirectionValue, attributeCount);
+
+            uint16_t loopTimeValue = 0x0019; // Default loop time value according to spec
+            Attributes::ColorLoopTime::Get(endpoint, &loopTimeValue);
+            AddAttributeValuePair(pairs, Attributes::ColorLoopTime::Id, loopTimeValue, attributeCount);
+        }
+
+        if (ColorControlServer::Instance().HasFeature(endpoint, ColorControlServer::Feature::kColorTemperature))
+        {
+            uint16_t temperatureValue = 0x00FA;
+            Attributes::ColorTemperatureMireds::Get(endpoint, &temperatureValue);
+            AddAttributeValuePair(pairs, Attributes::ColorTemperatureMireds::Id, temperatureValue, attributeCount);
+        }
+
+        uint8_t modeValue = ColorControlServer::ColorMode::COLOR_MODE_CIE_XY;
+        Attributes::EnhancedColorMode::Get(endpoint, &modeValue);
+        AddAttributeValuePair(pairs, Attributes::EnhancedColorMode::Id, modeValue, attributeCount);
+
+        app::DataModel::List<AttributeValuePair> attributeValueList(pairs, attributeCount);
+
+        return EncodeAttributeValueList(attributeValueList, serializedBytes);
+    }
+
+    /// @brief Default EFS interaction when applying scene to the ColorControl Cluster
+    /// @param endpoint target endpoint
+    /// @param cluster  target cluster
+    /// @param serialisedBytes Data from nvm
+    /// @param timeMs transition time in ms
+    /// @return CHIP_NO_ERROR if value as expected, CHIP_ERROR_INVALID_ARGUMENT otherwise
+    CHIP_ERROR ApplyScene(EndpointId endpoint, ClusterId cluster, const ByteSpan & serializedBytes,
+                          scenes::TransitionTimeMs timeMs) override
+    {
+        app::DataModel::DecodableList<Scenes::Structs::AttributeValuePair::DecodableType> attributeValueList;
+
+        ReturnErrorOnFailure(DecodeAttributeValueList(serializedBytes, attributeValueList));
+
+        size_t attributeCount = 0;
+        uint32_t attributeBuffer[kColorControlMaxScenableAttributes];
```

Comment:
So this contains random values, right?  What if some of the values are missing from the attributeValueList?

Reply:
Got rid of the buffer and used default values where storage was needed.